File: common/src/main/java/com/ultreon/mods/advanceddebug/api/IAdvancedDebug.java
package com.ultreon.mods.advanceddebug.api;

import com.ultreon.mods.advanceddebug.api.client.menu.IDebugGui;
import com.ultreon.mods.advanceddebug.api.client.registry.IFormatterRegistry;
import com.ultreon.mods.advanceddebug.client.Config;
import io.github.xypercode.craftyconfig.CraftyConfig;

import java.lang.reflect.InvocationTargetException;

public interface IAdvancedDebug {
    String MOD_ID = "advanced_debug";

    static IAdvancedDebug get() {
        try {
            return (IAdvancedDebug) Class.forName("com.ultreon.mods.advanceddebug.AdvancedDebug").getDeclaredMethod("getInstance").invoke(null);
        } catch (IllegalAccessException | InvocationTargetException | NoSuchMethodException |
                 ClassNotFoundException e) {
            throw new IllegalStateException("Can't get instance object of the Advanced Debug Mod.");
        }
    }

    CraftyConfig getConfig();

    IDebugGui getGui();

    IFormatterRegistry getFormatterRegistry();

    String getModId();

    @Deprecated(forRemoval = true)
    default boolean isSpacedNamespace() {
        return Config.spacedNamespaces;
    }

    @Deprecated(forRemoval = true)
    default boolean isSpacedEnumConstants() {
        return Config.spacedEnumConstants;
    }

    @Deprecated(forRemoval = true)
    default boolean enableBubbleBlasterID() {
        return Config.enableBubbleBlasterId;
    }

    default boolean isNamespaceSpaced() {
        return Config.spacedNamespaces;
    }

    default boolean isEnumConstantsSpaced() {
        return Config.spacedEnumConstants;
    }

    default boolean isBubbleBlasterIdEnabled() {
        return Config.enableBubbleBlasterId;
    }
}

File: common/src/main/kotlin/com/ultreon/mods/advanceddebug/api/IAdvancedDebug.kt
package com.ultreon.mods.advanceddebug.api

import com.ultreon.mods.advanceddebug.api.client.menu.IDebugGui
import com.ultreon.mods.advanceddebug.api.client.registry.IFormatterRegistry
import java.lang.reflect.InvocationTargetException

interface IAdvancedDebug {
    val gui: IDebugGui
    val formatterRegistry: IFormatterRegistry
    val isSpacedNamespace: Boolean
    val isSpacedEnumConstants: Boolean
    fun enableBubbleBlasterID(): Boolean

    companion object {
        fun get(): IAdvancedDebug {
            return try {
                Class.forName("AdvancedDebug").getDeclaredMethod("getInstance")(null) as IAdvancedDebug
            } catch (e: IllegalAccessException) {
                throw IllegalStateException("Can't get instance object of the Advanced Debug Mod.")
            } catch (e: InvocationTargetException) {
                throw IllegalStateException("Can't get instance object of the Advanced Debug Mod.")
            } catch (e: NoSuchMethodException) {
                throw IllegalStateException("Can't get instance object of the Advanced Debug Mod.")
            } catch (e: ClassNotFoundException) {
                throw IllegalStateException("Can't get instance object of the Advanced Debug Mod.")
            }
        }
    }
}